packages/flutter/lib/src/material/dropdown_menu_form_field.dart

```diff
@@ -0,0 +1,235 @@
+    super.forceErrorText,
+  }) : super(
+         initialValue: initialSelection,
+         autovalidateMode: autovalidateMode,
+         builder: (FormFieldState<T> field) {
+           final _DropdownMenuFormFieldState<T> state = field as _DropdownMenuFormFieldState<T>;
+           void onSelectedHandler(T? value) {
+             field.didChange(value);
+             onSelected?.call(value);
+           }
+
+           return UnmanagedRestorationScope(
+             bucket: field.bucket,
+             child: DropdownMenu<T>(
+               restorationId: restorationId,
+               enabled: enabled,
+               width: width,
+               menuHeight: menuHeight,
+               leadingIcon: leadingIcon,
+               trailingIcon: trailingIcon,
+               label: label,
+               hintText: hintText,
+               helperText: helperText,
+               errorText: state.errorText,
+               selectedTrailingIcon: selectedTrailingIcon,
+               enableFilter: enableFilter,
+               enableSearch: enableSearch,
+               keyboardType: keyboardType,
+               textStyle: textStyle,
+               textAlign: textAlign,
+               inputDecorationTheme: inputDecorationTheme,
+               menuStyle: menuStyle,
+               controller: controller,
+               initialSelection: state.value,
+               onSelected: onSelectedHandler,
+               focusNode: focusNode,
+               requestFocusOnTap: requestFocusOnTap,
+               expandedInsets: expandedInsets,
+               alignmentOffset: alignmentOffset,
+               filterCallback: filterCallback,
+               searchCallback: searchCallback,
+               inputFormatters: inputFormatters,
+               closeBehavior: closeBehavior,
+               dropdownMenuEntries: dropdownMenuEntries,
+               maxLines: maxLines,
+               textInputAction: textInputAction,
+             ),
+           );
+         },
+       );
+
+  /// The callback is called when a selection is made.
+  ///
+  /// Defaults to null. If null, only the text field is updated.
+  final ValueChanged<T?>? onSelected;
+
+  /// Controls the text being edited.
+  ///
+  /// If null, this widget will create its own [TextEditingController].
+  final TextEditingController? controller;
+
+  /// Descriptions of the menu items in the [DropdownMenuFormField].
+  ///
+  /// This is a required parameter. It is recommended that at least one [DropdownMenuEntry]
+  /// is provided. If this is an empty list, the menu will be empty and only
+  /// contain space for padding.
+  final List<DropdownMenuEntry<T>> dropdownMenuEntries;
+
+  @override
+  FormFieldState<T> createState() => _DropdownMenuFormFieldState<T>();
+}
+
+class _DropdownMenuFormFieldState<T> extends FormFieldState<T> {
+  DropdownMenuFormField<T> get _dropdownMenuFormField => widget as DropdownMenuFormField<T>;
+
+  RestorableTextEditingController? _restorableController;
```

Comment:
Let me know if I'm understanding this right: You are restoring the value of the TextEditingController, but you're only using that to find the correct selection and not to restore the state of the field?

Reply:
Exactly, I tried to use it also to restore the selection but I hit some difficulties with the way DropdownMenu interact with it TextEditingController.